Development

tepavcevic opened the pull request:
Progress so far
Fetching data with useEffect from components. Separated fetching api data to a different file. Implemented fetching with url and options in order to fetch pages and items usinga single function. Extracted api url to a constant to be used across components. Minimal styling so far, not a priority at the moment.

tepavcevic committed:
Fetching api
Added pagination and fetching api
Very basic styling so far

tepavcevic committed:
Basic routing
Installing router and setting up routes for homepage and id pages
Installing prettier and config file

tepavcevic committed:
Pagination with react-router
Uninstaling react-pagination, and implementing it with react-router
and query parameters

tepavcevic committed:
React router integration with react paginate
Improved logic for data fetching in a separate file, as well as url to a
separate constant
Since api metadata is unavailable, number of items put in a constant
Number of pages calculated wrt that constant

vlado92 commented on src/constants/apiNumOfItems.js:
I would suggest to have single constants file which will export these constants

vlado92 commented on src/pages/beer-info/Index.jsx:
```diff
@@ -0,0 +1,60 @@
+      setBeer(data);
+    };
+
+    fetchBeer();
```
You can move all logic of fetchBeer to useEffect itself, no need to separate

vlado92 commented on src/pages/beer-info/Index.jsx:
```diff
@@ -0,0 +1,60 @@
+  useEffect(() => {
+    const fetchBeer = async () => {
+      const data = await fetchFromParams(BREWDOG_API, `/${id}`);
+      setBeer(data);
```
Did you account for fact data can be undefined? (if API request fails for any reason)

vlado92 replied:
Also, if this api returns array, I would suggest to do `setBeer(data[0])` because it is easier to manipulate later

vlado92 commented on src/domain/fetchAPI/fetchFromParams.js:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    const result = await response.json();
+    return result;
```
I would suggest to throw errors in case of error, and not silently fail

tepavcevic committed:
Making requested changes
Tweaking fetching data in useEffects
Moving constants to a single file
Better error handling

vlado92 commented on src/pages/home/Index.jsx:
```diff
@@ -2,8 +2,8 @@ import { useEffect, useState } from 'react';
 
-import { BREWDOG_API } from '../../constants/apiUrl';
-import { NUM_OF_ITEMS } from '../../constants/apiNumOfItems';
+import { BREWDOG_API } from '../../constants/constants';
```
Why did you not simplify this as

```import { BREWDOG_API, NUM_OF_ITEMS } from '../../constants/constants';```

tepavcevic committed:
Import fix
